### ISSUE TITLE
socket activated service enters failed state when client goes away too fast

### ISSUE THREAD

**Issue by mbiebl** [systemd/systemd#7172]
This was reported as a downstream bug report at
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=879603

It looks like systemd has a race condition where instead of giving up
gracefully when a client goes away, it enters a bogus failure mode.
```
● vsftpd-security@9330-149.20.4.14:21-X.X.X.X:Y.service - vsftpd security
   Loaded: loaded (/etc/systemd/system/vsftpd-security@.service; static;
vendor preset: enabled)
   Active: failed (Result: resources)

Oct 22 18:48:41 mirror-isc systemd[1]:
vsftpd-security@9330-149.20.4.14:21-X.X.X.X:Y.service: Failed to run
'start' task: Transport endpoint is not connected
Oct 22 18:48:41 mirror-isc systemd[1]: Failed to start vsftpd security.
Oct 22 18:48:41 mirror-isc systemd[1]:
vsftpd-security@9330-149.20.4.14:21-X.X.X.X:Y.service: Unit entered
failed state.
Oct 22 18:48:41 mirror-isc systemd[1]:
vsftpd-security@9330-149.20.4.14:21-X.X.X.X:Y.service: Failed with
result 'resources'.
```
service file
```
# /etc/systemd/system/vsftpd-security@.service
[Unit]
Description=vsftpd security

[Service]
ExecStart=-/usr/sbin/vsftpd /etc/vsftpd-security.conf
StandardInput=socket
StandardError=journal
CapabilityBoundingSet=CAP_SYS_CHROOT CAP_SETUID CAP_SETGID
PrivateDevices=true
ProtectHome=true
ProtectSystem=full

# /etc/systemd/system/vsftpd-security.socket
[Unit]
Description=vsftpd security (socket)

[Socket]
ListenStream=149.20.4.14:21
ListenStream=[2001:4f8:1:c::14]:21
Accept=true
FreeBind=true
MaxConnections=200

[Install]
WantedBy=sockets.target
```

What I expect is for the service not to end up in a failed state for
something which, as far as I can tell, is not an error condition.  I
expect that if a remote client goes away before systemd can exec vsftpd
(which is what the error messages hint at; if that's not what's going
on, then there's a separate bug that the errors are misleading), it
should treat that as business as usual, and basically ignore it.

https://github.com/systemd/systemd/issues/1742 looks kind of similar to this bug, but that one's on ExecStop whereas this here is about start.

**Comment by poettering** [comment]
which systemd version is this about? i presume it contains 7dd736abec and friends?

any chance you can run `strace -p1 -s 500` before this, then trigger this, so that we can see where precisely the ENOTCONN is generated

**Comment by mbiebl** [comment]
It's version v232 which contains 7dd736a.

As for the strace, I'll forward that question to the original bug reporter.

**Comment by mbiebl** [comment]
Reply from the bug reporter: "Unfortunately this isn't something I can reproduce at will, it just
shows up from time to time."

**Comment by poettering** [comment]
Ah, reading 7dd736abecd8048843d117de56c65bbf489a6f4f again, it actually explains the issue at hand here... I figure we should deal with this in a smarter way.

**Comment by poettering** [comment]
Fix waiting in #7329.
